airflow-core/src/airflow/dag_processing/bundles/manager.py

```diff
@@ -124,12 +174,45 @@ def parse_config(self) -> None:
     @provide_session
     def sync_bundles_to_db(self, *, session: Session = NEW_SESSION) -> None:
         self.log.debug("Syncing DAG bundles to the database")
+
+        def _signed_template(new_template_: str | None, bundle_name: str) -> str | None:
+            if new_template_:
+                if not _is_safe_bundle_url(new_template_):
+                    self.log.warning(
+                        "Bundle %s has unsafe URL template '%s', skipping URL update",
+                        bundle_name,
+                        new_template_,
+                    )
+                    new_template_ = None
+                else:
+                    # Sign the URL for integrity verification
+                    new_template_ = _sign_bundle_url(new_template_, bundle_name)
+                    self.log.debug("Signed URL template for bundle %s", bundle_name)
+            return new_template_
+
         stored = {b.name: b for b in session.query(DagBundleModel).all()}
+
         for name in self._bundle_config.keys():
+            # Update URL template and parameters if they've changed
+            bundle_instance = self.get_bundle(name)
+            new_template = bundle_instance.view_url_template()
+            new_params = self._extract_template_params(bundle_instance)
```

Comment:
I wonder if we can just move extraction of these two directly into the signed_template and call it:
`extract_and_sign_bundle_template(bundle_instance)` or even better move line 197 too. That way, it only takes bundle_name.